### PR TITLE
Fedora 30 is coming to town

### PR DESCRIPTION
**Pull request by marusak**
* [x] image-refresh fedora-29
 * [x] image-refresh fedora-30

General stuff:
- [x] ~~Booting is slow - investigate and make faster~~
- [x] Open the server in tmpdir

Failed tests:
- [x] testCockpitDesktop (check_connection.TestConnection)
- [x] testLocalSession (check_connection.TestConnection)
- [x] testCreate (check_machines_dbus.TestMachinesDBus)
- [x] testStoragePoolsCreate (check_machines_dbus.TestMachinesDBus)
- [x] testDialogTransition (check_realms.TestPackageInstall)
- [x] testBasic (check_storage_multipath.TestStorage)
- [x] testTime (check_system_info.TestSystemInfo)
- [x] testTimeServers (check_system_info.TestSystemInfo)

[original log](https://209.132.184.41:8493/logs/pull-11432-20190321-092828-bb4133f9-verify-fedora-30/log.html#217)

### CODE REVIEW

**Comment by cockpituous** [comment]
image-refresh in progress on centosci-tasks-2txqb.
Log: https://209.132.184.41:8493/logs/image-refresh-11432-20190320-154242/

**Comment by cockpituous** [comment]
image-refresh fedora-29 done: https://github.com/cockpituous/cockpit/commits/image-refresh-fedora-29-20190320-161212

**Comment by cockpituous** [comment]
image-refresh in progress on centosci-tasks-2txqb.
Log: https://209.132.184.41:8493/logs/image-refresh-11432-20190320-161212/

**Comment by cockpituous** [comment]
image-refresh in progress on bos-tasks-gv3x1.
Log: https://209.132.184.41:8493/logs/image-refresh-11432-20190320-162323/

**Comment by cockpituous** [comment]
image-refresh fedora-30 done: https://github.com/cockpituous/cockpit/commits/image-refresh-fedora-30-20190320-174949

**Comment by marusak** [comment]
@martinpitt last time image got lost you told me to ping you - so ping :)
F30 image cannot be found anywhere, can you help me with this or do I need to rebuild it?

**Comment by martinpitt** [comment]
@marusak: Sorry, this one is really gone :-/ Retriggering.

**Comment by cockpituous** [comment]
image-refresh in progress on centosci-tasks-wxgrr.
Log: https://209.132.184.41:8493/logs/image-refresh-11432-20190320-214747/

**Comment by cockpituous** [comment]
image-refresh fedora-30 done: https://github.com/cockpituous/cockpit/commits/image-refresh-fedora-30-20190320-221515

**Comment by martinpitt** [comment]
@marusak : Note that you still need to add a Closes #. Once f-30 becomes green, I suggest force-pushing without [no-test], so that all tests run (F-29 image is involved in quite a lot of them). Thanks!

**Comment by martinpitt** [comment]
The ovirt tests should be skipped  for F30, we deliberately don't build  cockpit-machines-ovirt for Fedora > 29 (see tools/cockpit.spec).

I downloaded the image to test something unrelated, and booting is achingly slow:
```
# systemd-analyze
Startup finished in 1.163s (kernel) + 2.349s (initrd) + 2min 44.214s (userspace) = 2min 47.727s 
multi-user.target reached after 2min 32.067s in userspace

# systemd-analyze blame
    2min 40.436s unbound-anchor.service
    1min 30.023s gssproxy.service
         58.297s libvirtd.service
         14.068s sshd.service
[... rest is reasonable ]
```
This is okay for a first version, but this should get some bug reports and masking of unnecessary services, to not inflate test runtime so much.

**Comment by marusak** [comment]
> The ovirt tests should be skipped for F30

will do.

> booting is achingly slow

Good catch. I will look at this and see if we can just disable them from autostart or what else needs to be done.

1st run: 11 tests failed: (ovirt are gonna be skipped, others rerun to see if flake or real issue)
16: testCockpitDesktop (check_connection.TestConnection)
21: testLocalSession (check_connection.TestConnection)
63: testCreate (check_machines_dbus.TestMachinesDBus)
78: testStoragePoolsCreate (check_machines_dbus.TestMachinesDBus)
154: testBasic (check_ovirt.TestOVirtMachines)
155: testHostToMaintenance (check_ovirt.TestOVirtMachines)
156: testVCPU (check_ovirt.TestOVirtMachines)
164: testDialogTransition (check_realms.TestPackageInstall)
205: testBasic (check_storage_multipath.TestStorage)
219: testTime (check_system_info.TestSystemInfo)
220: testTimeServers (check_system_info.TestSystemInfo)

**Comment by marusak** [comment]
> I downloaded the image to test something unrelated, and booting is achingly slow:
> ```
> # systemd-analyze
> Startup finished in 1.163s (kernel) + 2.349s (initrd) + 2min 44.214s (userspace) = 2min 47.727s 
> multi-user.target reached after 2min 32.067s in userspace
>
> # systemd-analyze blame
>     2min 40.436s unbound-anchor.service
>     1min 30.023s gssproxy.service
>          58.297s libvirtd.service
>          14.068s sshd.service
> [... rest is reasonable ]
> ```

I tried this on my machine and was not able to reproduce this with F-30 image.

    $ ./bots/image-download fedora-30
    $ ./bots/image-prepare fedora-30
    $ ./test/vm-run fedora-30

    # systemd-analyze
    Startup finished in 1.064s (kernel) + 1.605s (initrd) + 16.373s (userspace) = 19.043s 
    multi-user.target reached after 16.342s in userspace

    # systemd-analyze blame
          7.693s pmlogger.service
          3.056s pmcd.service
          2.802s pmie.service
          1.552s pmlogger_check.service
          1.287s NetworkManager-wait-online.service
          1.273s firewalld.service
          1.247s dev-mapper-fedora\x2droot.device
          1.224s systemd-udev-settle.service
          1.077s initrd-switch-root.service
          1.021s systemd-machined.service
           999ms lvm2-monitor.service
           923ms sssd.service
           842ms systemd-resolved.service
           727ms systemd-journald.service
           646ms systemd-logind.service
           568ms systemd-udevd.service
           507ms dracut-initqueue.service
           480ms lvm2-pvscan@252:2.service
           459ms abrtd.service
           392ms libvirtd.service
           385ms systemd-vconsole-setup.service
           327ms systemd-udev-trigger.service
           205ms chronyd.service

Can you write if you can reproduce it and if so, how you run the machine? (btw. 14s out of 19 takes pcp and friends, that would be worth disabling by default)

P.S. Lately my machine boots way to slow (still F-29). I did not really bother to investigate as I reboot just once in a while, but now I checked `systemd-analyze` and it takes more then 3 minutes :open_mouth: Main time takes (more than 20s) are `dracut-initqueue.service` and `systemd-cryptsetup@luks... .service`, but third is `unbound-anchor.service` with 12 seconds, so it may be somewhat related to your report.

**Comment by marusak** [comment]
There are 8 tests that fail stably. I will now go through them and will check what is the problem.

**Comment by marusak** [comment]
[One test fail](https://209.132.184.41:8493/logs/pull-11432-20190323-060303-797be752-verify-fedora-30/log.html#77), likely flake? Retrying.

**Comment by marusak** [comment]
Seems like flake. I will edit that server is served from tmpdir, rebuild images for F29 and F30 and when picked new images I'll let run all the tests and that hopefully should be it.

**Comment by cockpituous** [comment]
image-refresh in progress on bos-tasks-7fm0g.
Log: https://209.132.184.41:8493/logs/image-refresh-11432-20190323-114949/

**Comment by cockpituous** [comment]
image-refresh fedora-29 done: https://github.com/cockpituous/cockpit/commits/image-refresh-fedora-29-20190323-123030

**Comment by cockpituous** [comment]
image-refresh in progress on bos-tasks-3p2wv.
Log: https://209.132.184.41:8493/logs/image-refresh-11432-20190323-124040/

**Comment by cockpituous** [comment]
image-refresh in progress on centosci-tasks-0hzz4.
Log: https://209.132.184.41:8493/logs/image-refresh-11432-20190323-123131/

**Comment by cockpituous** [comment]
image-refresh in progress on bos-tasks-nj1c8.
Log: https://209.132.184.41:8493/logs/image-refresh-11432-20190323-123131/

**Comment by cockpituous** [comment]
image-refresh in progress on bos-tasks-7fm0g.
Log: https://209.132.184.41:8493/logs/image-refresh-11432-20190323-123131/

**Comment by cockpituous** [comment]
image-refresh in progress on bos-tasks-3hjf4.
Log: https://209.132.184.41:8493/logs/image-refresh-11432-20190323-124141/

**Comment by cockpituous** [comment]
Task failed: https://209.132.184.41:8493/logs/image-refresh-11432-20190323-124141/

**Comment by cockpituous** [comment]
image-refresh in progress on bos-tasks-5rb21.
Log: https://209.132.184.41:8493/logs/image-refresh-11432-20190323-145959/

**Comment by cockpituous** [comment]
Task failed: https://209.132.184.41:8493/logs/image-refresh-11432-20190323-145959/

**Comment by cockpituous** [comment]
image-refresh in progress on centosci-tasks-0hzz4.
Log: https://209.132.184.41:8493/logs/image-refresh-11432-20190323-150606/

**Comment by cockpituous** [comment]
image-refresh in progress on bos-tasks-5rb21.
Log: https://209.132.184.41:8493/logs/image-refresh-11432-20190323-151616/

**Comment by cockpituous** [comment]
image-refresh fedora-30 done: https://github.com/cockpituous/cockpit/commits/image-refresh-fedora-30-20190323-154949

File: test/common/testlib.py [hunk 1]
```diff
@@ -916,6 +916,10 @@ def check_journal_messages(self, machine=None):
             self.allowed_messages.append('audit: type=1400 audit(.*): avc:  denied  { search } for  .* comm="qemu-.* dev="proc" .*')
             self.allowed_messages.append('audit: type=1400 audit(.*): avc:  denied  { read } for  .* comm="qemu-.* dev="proc" .*')
 
+        if self.image in ['fedora-30']:
+            # Fedora 30 switched to dbus-broker
+            self.allowed_messages.append("dbus-daemon didn't send us a dbus address; not installed?.*")
```
Comment (martinpitt) [inline test/common/testlib.py]: This is hiding a bug, but that's ok for a follow-up.

**Pull request merged**